### PR TITLE
ref(ingest): Completely inline save_event into consumer for transactions

### PR DESCRIPTION
**Pull request by untitaker**
Follow up to #17191 

TODO:

- [x] replace eventmanager.save() calls with this

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -349,19 +349,25 @@ def clean(d):
...
-    def normalize(self):
+    def normalize(self, project_id=None):
         with metrics.timer("events.store.normalize.duration"):
-            self._normalize_impl()
+            self._normalize_impl(project_id=project_id)
+
+    def _normalize_impl(self, project_id=None):
+        if self._project and project_id and project_id != self._project.id:
+            raise RuntimeError(
+                "Initialized EventManager with one project ID and called save() with another one"
```
Comment (jan-auer) [inline src/sentry/event_manager.py]: Can we prevent this by not initializing EventManager anymore?

File: src/sentry/ingest/ingest_consumer.py [hunk 2]
```diff
@@ -84,25 +88,46 @@ def flush_batch(self, batch):
...
+            process_transactions_batch(transactions, projects)
 
     def shutdown(self):
         pass
 
 
+@metrics.wraps("ingest_consumer.process_transactions_batch")
+def process_transactions_batch(messages, projects):
+    events = []
+    for message in messages:
```
Comment (jan-auer) [inline src/sentry/ingest/ingest_consumer.py]: Do you want to time this loop or parts of it?

**Review by jan-auer — APPROVED** [review]

File: src/sentry/event_manager.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/event_manager.py]: yes, but that's a large diff. let's make smaller steps

File: src/sentry/ingest/ingest_consumer.py (hunk 2 above)
Comment (untitaker) [reply src/sentry/ingest/ingest_consumer.py]: it's just json decode so i am timing that now

**Comment by fpacifici** [comment]
Wait a minute. This is going to bypass Celery, which has impact on parallelism and imposes a tradeoff in number of partitions. Is the ops team onboard with the decision ?

**Comment by untitaker** [comment]
@fpacifici fwiw we're having a broader discussion about how to approach/verify perf improvements and I think one of the consequences is that we'll revert things like this. This was a mistake.

**Pull request merged**